db/data.py

```diff
@@ -359,6 +360,58 @@ def load_low_level(mbid, offset=0):
         return row[0]
 
 
+def load_many_low_level(recordings):
+    """Recordings is a list of tuples (mbid, offset).
+       Uses a single query method to collect low level data for multiple recordings.
+       
+       Steps:
+        - Creates dict structure as sampled below
+        - Subquery uses `row_number` with `partition by` attach offsets to recordings
+        - Outer query left joins lowlevel_json to select data
+        - recordings_info is used as a parameter to select recordings/offsets 
+        - recordings_info formatted as jsonb
+
+       Sample query response returned to api/v1/core.py get_data_for_multiple_recordings():
+
+       {"mbid1": {"offset1": {lowlevel_data},
+                  "offsetn": {lowlevel_data}}
+         ...
+         ...
+
+        "mbidn": {"offset1": {lowlevel_data}}
+       }
+    
+    """
+    recordings_info = {}
+    for mbid, offset in recordings:
+        recordings_info.setdefault(mbid, {})[offset] = {}
+
+    recordings_info_json = json.dumps(recordings_info, separators=(',', ':'))
+    
+    with db.engine.connect() as connection:
+        query = text(
+            """ SELECT gid, RowNum, data
```

Comment:
Could you elaborate further on how this would look? I was under the impression that select will be executed after the where clause, and thus an alias from select cannot be used in the where clause - the full expression must be used instead. When trying to use the alias, I see an error in which the column name is not found. The postgres docs support this: https://www.postgresql.org/docs/current/sql-select.html#SQL-SELECT-LIST

Reply:
That being said, I see an alternative to both. Below, in the inner query, I can select `ROW_NUMBER () OVER (PARTITION BY gid) -1 RowNum`, then the column RowNum (soon to be submission_offset) will start from 0 in the first place.